src/core/utils/slaac_address.cpp

```diff
@@ -234,7 +236,7 @@ void Slaac::Update(UpdateMode aMode)
                     slaacAddr->Clear();
                     memcpy(&slaacAddr->mAddress, &prefix.mPrefix, BitVectorBytes(prefix.mLength));
 
-                    slaacAddr->mPrefixLength  = prefix.mLength;
+                    slaacAddr->mPrefixLength  = config.mOnMesh ? prefix.mLength : 128;
```

Comment:
What if this prefix config is `off-mesh` but there is another same prefix afterwards but is `on-mesh`. I think we need to generate SLAAC address of prefix  of `prefix.mLength` rather than 128 for this prefix. 
Seems we need the full information of a prefix to generate the correct address?

Reply:
Added a check above to ensure we will generate a new address.